### PR TITLE
Move the ComponentConfig types to staging KEP to implementable

### PR DESCRIPTION
**Pull request by luxas**
Some minor changes to the initial KEP after further discussions in the sig-api-machinery Slack channel.
Builds on top of https://github.com/kubernetes/community/pull/2354 and marks it as `implementable`.
We have now submitted and merged the ~5 first PRs of this effort, with more PRs planned and in the reviewing stage. We hope to get all of this implemented by v1.12 code freeze.

The additions to this version is that we've decided on a defaulting story, and know a bit more about the actual implementation plan, and know what tests to write up for the MVP.

Again, mentioning the SIGs that are affected by this proposal:

@kubernetes/sig-cluster-lifecycle-pr-reviews @kubernetes/sig-api-machinery-pr-reviews @kubernetes/sig-architecture-api-reviews @kubernetes/sig-node-pr-reviews @kubernetes/sig-cloud-provider-pr-reviews @kubernetes/sig-scheduling-pr-reviews @kubernetes/sig-network-pr-reviews

/assign @thockin @jbeda @liggitt @bgrant0607 @mtaufen @wojtek-t

### CODE REVIEW

**Comment by hogepodge** [comment]
From sig cloud-provider pov, these updates look fine to me. I'll defer to anything that @andrewsykim or @jagosan pick up on, though.

File: keps/sig-cluster-lifecycle/0014-20180707-componentconfig-api-types-to-staging.md [hunk 1]
```diff
@@ -304,13 +318,15 @@ Dep supports the import of sub-packages without inheriting dependencies from out
...
+* Remove as many unnecessary references to `pkg/apis/componentconfig` from the rest of the core repo as possible
+* Make the types in `pkg/apis/componentconfig` reuse the newly-created types in `k8s.io/apiserver` and `k8s.io/apimachinery `.
+* Start with the scheduler as the first component to be moved out.
+    * One PR for moving `KubeSchedulerConfiguration` to `staging/src/k8s.io/kube-scheduler/config/v1alpha1/types.go`, and the internal type to `pkg/scheduler/apis/config/types.go`.
+        * Set up the conversion for the external type by creating the package `pkg/scheduler/apis/config/v1alpha1`, without `types.go`, like how `k8s.io/api` is set up.
+        * This should be a pure code move.
 * Set up staging publishing bot (async, non-critical)
-* The kubelet, kube-proxy and controller-manager follow, each one independently.
+* The kubelet, kube-proxy and kube-controller-manager types follow, each one independently.
```
Comment (sttts) [inline keps/sig-cluster-lifecycle/0014-20180707-componentconfig-api-types-to-staging.md]: kube+cloud-controller-manager

**Comment by sttts** [comment]
Looks good.

**Pull request merged**